### PR TITLE
Sprint 3/4 closure

### PR DESCRIPTION
**Pull request by z4y4ts**
New features:
- Default transcripts. Allows to fetch transcripts from the platform and store them into XBlock.
- Brightcove auto-quality and content encryption.

Various bugfixes and improvements

### CODE REVIEW

**Comment by auraz** [comment]
What is test coverage?
Fix commit names according to standards.

File: .coveralls.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+repo_token: St0myP2WKubt2SrFuyQDZI7Y1gsxDIiiX
```
Comment (auraz) [inline .coveralls.yml]: Just curious, it is for which repo?

File: .codeclimate.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+engines:
```
Comment (auraz) [inline .codeclimate.yml]: Please add docstring, why this file is needed.

File: .editorconfig [hunk 3]
```diff
@@ -10,3 +10,6 @@ trim_trailing_whitespace = true
 
 [*.yml]
 indent_size = 2
+
+[Makefile]
+indent_style = tab
```
Comment (auraz) [inline .editorconfig]: Why this file is in the repo?

Comment (auraz) [inline .editorconfig]: TAB?

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,33 @@
+.PHONY=all,quality,test
```
Comment (auraz) [inline Makefile]: Please do not use makefile. 
Use paver at least.

File: video_xblock/backends/brightcove.py [hunk 5]
```diff
@@ -67,7 +86,185 @@ def get_frag(self, **context):
...
+                        'lang': 'en',
+                        'label': 'English',
+                        'url': 'learning-services-media.brightcove.com/captions/bc_smart_ja.vtt'
+                    },
+                    # ...
+                ]
+            message (str): message for a user on default transcripts fetching.
+        """
+        video_id = kwargs.get('video_id')
+        account_id = kwargs.get('account_id')  # TODO add handling: default account_id
```
Comment (auraz) [inline video_xblock/backends/brightcove.py]: Fix TODO?

File: video_xblock/backends/brightcove.py [hunk 6]
```diff
@@ -67,7 +86,185 @@ def get_frag(self, **context):
...
+        else:
+            try:
+                message = str(text[0].get('message'))
+            except AttributeError:
+                message = 'No timed transcript may be fetched from a video platform. API response status: {}'.\
+                    format(str(data.status_code))
+        return default_transcripts, message
+
+    def download_default_transcript(self, url):  # pylint: disable=unused-argument
+        # TODO: implement
```
Comment (auraz) [inline video_xblock/backends/brightcove.py]: Fix TODO, add docstring.

File: video_xblock/backends/wistia.py [hunk 7]
```diff
@@ -69,3 +92,123 @@ def get_frag(self, **context):
...
+            message = 'For now, video platform doesn\'t have any timed transcript for this video.'
+        # If a video does not exist, the response will be an empty HTTP 404 Not Found.
+        # Reference: https://wistia.com/doc/data-api#captions_index
+        elif data.status_code == 404:
+            message = "Wistia video {video_id} doesn't exist.".format(video_id=str(video_id))
+
+        return default_transcripts, message
+
+    def download_default_transcript(self, url):  # pylint: disable=unused-argument
+        # TODO: implement
```
Comment (auraz) [inline video_xblock/backends/wistia.py]: Fix TODO, add docstring.

File: video_xblock/backends/youtube.py [hunk 8]
```diff
@@ -60,3 +83,130 @@ def get_frag(self, **context):
...
+                text = element.text
+                end = start + duration
+                if text:
+                    # Start and end should be ints representing the millisecond timestamp.
+                    sub_starts.append(int(start * 1000))
+                    sub_ends.append(int((end + 0.0001) * 1000))
+                    sub_texts.append(text.replace('\n', ' '))
+
+        sub_dict = {'start': sub_starts, 'end': sub_ends, 'text': sub_texts}  # pylint: disable=unused-variable
+        # TODO implement conversion of sub_dict to WebVVT format
```
Comment (auraz) [inline video_xblock/backends/youtube.py]: Fix TODO

File: video_xblock/fields.py [hunk 9]
```diff
@@ -0,0 +1,123 @@
+"""
+RelativeTive field back-ported from xmodule.fields to avoid import error and travis testing complication
```
Comment (auraz) [inline video_xblock/fields.py]: put link to original file here

File: video_xblock/static/js/toggle-button.js [hunk 10]
```diff
@@ -4,147 +4,159 @@
...
-      var tracks = this.player_.textTracks();
-      for (var i = 0; i < tracks.length; i++) {
-        var track = tracks[i];
-        if (track.kind === 'captions') {
-          this.player_.captionsLanguage = el.dataset.lang;
-          track.mode = track.language === this.player_.captionsLanguage ? 'showing': 'disabled';
+    'use strict';
+
+    var MenuItem = videojs.getComponent('MenuItem');
+    var ToggleMenuItem = videojs.extend(MenuItem, {
```
Comment (auraz) [inline video_xblock/static/js/toggle-button.js]: Add docstring  for every  function in all JS files.

File: video_xblock/utils.py [hunk 11]
```diff
@@ -0,0 +1,10 @@
+"""
+This module contains video xblock helpers.
+"""
+
+
+def ugettext(text):
```
Comment (auraz) [inline video_xblock/utils.py]: Why we need it?

File: .coveralls.yml (hunk 1 above)
Comment (z4y4ts) [inline .coveralls.yml]: For this :)
I've got the token from Coveralls

File: .editorconfig (hunk 3 above)
Comment (z4y4ts) [inline .editorconfig]: This file here to make it easier to follow indentation conventions.

And yes, tabs. Makefiles are special.

File: Makefile (hunk 4 above)
Comment (z4y4ts) [inline Makefile]: Why not Makefile?

File: video_xblock/utils.py (hunk 11 above)
Comment (z4y4ts) [inline video_xblock/utils.py]: @TamoshaytisV can you please answer?

Comment (OPersian) [inline video_xblock/utils.py]: This was to fix pep8 violation: `E731 do not assign a lambda expression, use a def`

File: video_xblock/backends/brightcove.py (hunk 5 above)
Comment (OPersian) [inline video_xblock/backends/brightcove.py]: Fixed with PR #104

File: video_xblock/backends/brightcove.py (hunk 6 above)
Comment (OPersian) [inline video_xblock/backends/brightcove.py]: Fixed with PR #104

File: video_xblock/backends/wistia.py (hunk 7 above)
Comment (OPersian) [inline video_xblock/backends/wistia.py]: Fixed with PR #104

File: video_xblock/backends/youtube.py (hunk 8 above)
Comment (OPersian) [inline video_xblock/backends/youtube.py]: Fixed with PR #104

File: Makefile (hunk 4 above)
Comment (z4y4ts) [inline Makefile]: Discussed this. Leaving as is for now.

File: video_xblock/static/js/toggle-button.js (hunk 10 above)
Comment (OPersian) [inline video_xblock/static/js/toggle-button.js]: Increased docstring coverage for all the JS files. As discussed with @z4y4ts, docstrings will be added to all the functions in the next PR (considered as a separate task), as some JS files represent big bunches of functionality, e.g. [ studio_edit.js ](https://github.com/raccoongang/xblock-video/blob/feature/HLSe/video_xblock/static/js/studio-edit.js)

**Pull request merged**